src/modal/modal-ref.ts

```diff
@@ -40,6 +67,55 @@ export class NgbModalRef {
 	private _hidden = new Subject<void>();
 	private _resolve: (result?: any) => void;
 	private _reject: (reason?: any) => void;
+	private _windowAttributes = [
+		'animation',
+		'ariaLabelledBy',
+		'ariaDescribedBy',
+		'backdrop',
+		'centered',
+		'fullscreen',
+		'keyboard',
+		'scrollable',
+		'size',
+		'windowClass',
+		'modalDialogClass',
+	];
+	private _backdropAttributes = ['animation', 'backdropClass'];
+
+	private _applyWindowOptions(windowInstance: NgbModalWindow, options: NgbModalOptions): void {
+		this._windowAttributes.forEach((optionName: string) => {
+			if (isDefined(options[optionName])) {
+				windowInstance[optionName] = options[optionName];
+			}
+		});
+	}
+
+	private _applyBackdropOptions(backdropInstance: NgbModalBackdrop, options: NgbModalOptions): void {
+		this._backdropAttributes.forEach((optionName: string) => {
+			if (isDefined(options[optionName])) {
+				backdropInstance[optionName] = options[optionName];
+			}
+		});
+	}
+
+	update(
```

Comment:
Documentation missing for this method. Once you add comment, it should be documented here (https://ng-bootstrap.github.io/#/components/modal/api#NgbModalRef) alongside the `close` and `dismiss` methods

Reply:
Done